pages/hold/request/[id].tsx

```diff
@@ -0,0 +1,340 @@
+import type { ItemMetadata } from "../../../src/types/itemTypes"
+
+interface HoldRequestPropsType {
+  discoveryBibResult: DiscoveryBibResult
+  discoveryItemResult: DiscoveryItemResult
+  isAuthenticated?: boolean
+}
+
+/**
+ * The Hold Request page renders a form to place a hold on an item.
+ * It renders some basic info about the bib and item based on the bib and item ids in the url,
+ * including pickup location options rendered as radio buttons in the form.
+ */
+export default function HoldRequestPage({
+  discoveryBibResult,
+  discoveryItemResult,
+  isAuthenticated,
+}: HoldRequestPropsType) {
+  const metadataTitle = `Item Request | ${SITE_NAME}`
+
+  const bib = new Bib(discoveryBibResult)
+  const item = new Item(discoveryItemResult, bib)
+  const holdId = `${item.bibId}-${item.id}`
+
+  const [alert, setAlert] = useState(false)
+  const [formPosting, setFormPosting] = useState(false)
+  const notificationRef = useRef<HTMLDivElement>()
+  const { onOpen, setItemMetadata } = useContext(FeedbackContext)
+
+  const router = useRouter()
+  const isLoading = useLoading()
+
+  useEffect(() => {
+    if (alert && notificationRef.current) {
+      notificationRef.current.focus()
+    }
+  }, [alert])
+
+  const handleSubmit = async (e) => {
+    e.preventDefault()
+
+    try {
+      setFormPosting(true)
+      const response = await fetch(`${BASE_URL}/api/hold/request/${holdId}`, {
+        method: "POST",
+        // TODO: serialize form data
+        body: "",
+      })
+      const responseJson = await response.json()
+
+      if (response.status !== 200) {
+        console.error("Error in hold  request api response", responseJson.error)
+        setAlert(true)
+        setFormPosting(false)
+        return
+      }
+
+      // Success state
+      await router.push(`${PATHS.HOLD_CONFIRMATION}/${holdId}`)
+      setFormPosting(false)
+    } catch (error) {
+      console.error("Error in hold  request api response", error)
+      setAlert(true)
+      setFormPosting(false)
+    }
+  }
+
+  const onContact = (metadata: ItemMetadata) => {
+    setItemMetadata(metadata)
+    onOpen()
+  }
+
+  return (
+    <>
+      <Head>
+        <meta property="og:title" content={metadataTitle} key="og-title" />
+        <meta
+          property="og:site_name"
+          content={metadataTitle}
+          key="og-site-name"
+        />
+        <meta name="twitter:title" content={metadataTitle} key="tw-title" />
+        <title key="main-title">{metadataTitle}</title>
+      </Head>
+      <Layout isAuthenticated={isAuthenticated} activePage="hold">
+        {/* Always render the wrapper element that will display the
+          dynamically rendered notification for focus management */}
+        <Box tabIndex={-1} ref={notificationRef}>
+          {alert && (
+            <Banner
+              type="negative"
+              heading="Request failed"
+              data-testid="hold-request-error"
+              sx={{ a: { color: "ui.link.primary" } }}
```

Comment:
This should use the default error color for the link. CC @charmingduchess 
Check the documentation in Storybook and the "Design" tab: https://nypl.github.io/nypl-design-system/reservoir/v3/?path=/story/components-feedback-banner--with-controls 

<img width="733" alt="Screenshot 2024-10-22 at 5 12 39 PM" src="https://github.com/user-attachments/assets/8b969510-f556-4a88-a485-4e01552ecdc6">
<img width="1084" alt="Screenshot 2024-10-22 at 5 13 08 PM" src="https://github.com/user-attachments/assets/da9d4c60-7b4f-4b77-9cae-b17b978a141f">

Reply:
Done! had to add an override for the button styles